src/thor/alternates.cc

```diff
@@ -58,7 +58,8 @@ void filter_alternates_by_stretch(std::vector<CandidateConnection>& connections)
 
 // Limited Sharing. Compare duration of edge segments shared between optimal path and
 // candidate path. If they share more than kAtMostShared throw out this alternate.
-bool validate_alternate_by_sharing(GraphReader& graphreader,
+// Note that you should recover all shortcuts before call this function.
+bool validate_alternate_by_sharing(GraphReader& /*graphreader*/,
```

Comment:
can you actually just remove the graphreader argument completely, now that we dont ahve to recover shortcuts its not used.